### PR TITLE
Update main.cpp

### PR DESCRIPTION
**Pull request by livintune**
Tidied up the code a fair bit - there was an issue with the loops where the actual motor movement was being controlled outside of the while loop for line following - redColourRecognised would always be false, so it would have never broken out of the loop.

### CODE REVIEW

**Comment by livintune** [comment]
Fixed boolean to bool

File: main.cpp [hunk 1]
```diff
@@ -62,7 +62,7 @@ int main(){
     
         for(int i = 0; i < 16; i++){
             whiteValue = get_pixel(i*20, 120, 3);
-            if(whiteValue > 127){
+            if(whiteValue > 100){    //127 was too high - white values started at about 110
```
Comment (GregHyson) [inline main.cpp:50]: Agreed

File: main.cpp [hunk 2]
```diff
@@ -112,10 +112,11 @@ int main(){
         }
         
         if(reversed == false){                            // If the robot has just reversed we dont want it to then go off the previous picture which must have had no white in it, therefore we want it to take another picture and try moving again
+            
             float centerofWhite = (startOfWhite + endOfWhite)/2;
             float errorValue = centerOfWhite - 7.5;           // 7.5 is the center, therefore when it is equal the error value will be zero
             float d = 0;                                      // We need to test this variable as different sizes in order to find a size that corrects/ prevents oscillation.
-    
+            //here we want to calculate PID values - simply adding on a number will do nothing to change the oscillations
```
Comment (GregHyson) [inline main.cpp]: Yeh thats fine, your territory here and it looks right

File: main.cpp [hunk 3]
```diff
@@ -126,6 +127,7 @@ int main(){
             Sleep(0, 500000);
         }
         
+        //why are we setting these?
```
Comment (GregHyson) [inline main.cpp]: I would need to look over the entirety of the code if I wanted to know what I was thinking but its the kinda thing you are probably correct about me being incorrect.

**Pull request merged**